compiler/rustc_lexer/src/lib.rs

```diff
@@ -487,11 +489,16 @@ impl Cursor<'_> {
         RawIdent
     }
 
-    fn ident(&mut self) -> TokenKind {
+    fn ident_or_bad_prefix(&mut self) -> TokenKind {
         debug_assert!(is_id_start(self.prev()));
         // Start is already eaten, eat the rest of identifier.
         self.eat_while(is_id_continue);
-        Ident
+        // Good prefixes must have been handled eariler. So if
+        // we see a prefix here, it is definitely a bad prefix.
+        match self.first() {
+            '#' | '"' | '\'' => BadPrefix,
+            _ => Ident,
+        }
```

Comment:
Can you add a test to verify that the tokenization in Edition 2015 and 2018 doesn't change? I'm not sure the following will remain valid (see [this comment](https://github.com/rust-lang/rust/pull/84037#issuecomment-816798956)):
```

Also, can you split this function into a `fn ident()` and a separate `fn ident_or_reserved_prefix`? That would make the big match above less confusing.
quote! {
    #name = #kind#value
}
```

Reply:
Yeah, thanks for pointing this issue. I'll check it later.

Reply:
Added that exact snippet in the test.